firedrake/adjoint/blocks.py

```diff
@@ -146,6 +148,76 @@ def _ad_nlvs_replace_forms(self):
         self._ad_assign_coefficients(problem.F)
         self._ad_assign_coefficients(problem.J)
 
+    def prepare_evaluate_adj(self, inputs, adj_inputs, relevant_dependencies):
+        dJdu = adj_inputs[0]
+
+        F_form = self._create_F_form()
+
+        dFdu_form = self.adj_F
+        dJdu = dJdu.copy()
```

Comment:
What is `adj_inputs`? Could it be that you need a deepcopy here?

Reply:
`adj_inputs` are the adjoint values associated with the output variables (which become inputs as we traverse backwards through the tree in the adjoint). Not entirely sure why a copy is needed here but it's the same it was doing previously in the prepare_evaluate_adj of GenericSolveBlock that is being overloaded here.